test/plugins/adonis.spec.js

```diff
@@ -0,0 +1,327 @@
+          const config = new Config()
+          config.set('app.http.jsonpCallback', 'callback')
+          return new Response(this.request, config)
+        }, true)
+
+        setupResolver()
+        ioc.autoload(path.join(__dirname, 'app'), 'App')
+
+        return agent.load(plugin, 'adonis').then(() => {
+        })
+      })
+
+      after(() => {
+        return agent.close()
+      })
+
+      beforeEach(() => {
+        tracer = require(reqPath)
+        this.exception = Exception
+        this.server = new Server(Context, Route, new Logger(), this.exception)
+        this.server.bindExceptionHandler()
+
+        ioc.fake('Adonis/Exceptions/BaseExceptionHandler', () => {
+          return new BaseExceptionHandler()
+        })
+
+        return getPort().then(newPort => {
+          port = newPort
+        })
+      })
+
+      afterEach(() => {
+        RouteStore.clear()
+        ioc.restore()
+        this.exception.clear()
+        return this.server.close()
+      })
+
+      it('should do automatic instrumentation on routers', done => {
+        Route.get('/api/user', async function ({ response }) {
+          return response.send('GET')
+        })
+
+        agent.use(traces => {
+          const spans = traces[0]
+
+          expect(spans[0]).to.have.property('service', 'test')
+          expect(spans[1]).to.have.property('name', 'GET')
+          expect(spans[1].meta).to.have.property('component', 'adonis')
+          expect(spans[1].meta).to.have.property('span.kind', 'server')
+          expect(spans[1].meta).to.have.property('http.url', `http://localhost:${port}/api/user`)
+          expect(spans[1].meta).to.have.property('http.method', 'GET')
+          expect(spans[1].meta).to.have.property('http.status_code', '200')
+        }).then(done)
+          .catch(done)
+
+        this.server.listen('localhost', port, () => {
+          axios
+            .get(`http://localhost:${port}/api/user`)
+            .catch(done)
+        })
+      })
+
+      it('should properly tag errors', done => {
+        Route.get('/api/user', async function () {
+            throw new Error('custom error')
+          })
+        
+        agent.use(traces => {
+          const spans = traces[0]
+
+          expect(spans[0]).to.have.property('service', 'test')
+
+          expect(spans[1]).to.have.property('name', 'GET')
+          expect(spans[1].meta).to.have.property('component', 'adonis')
+          expect(spans[1].meta).to.have.property('span.kind', 'server')
+          expect(spans[1].meta).to.have.property('http.url', `http://localhost:${port}/api/user`)
+          expect(spans[1].meta).to.have.property('http.method', 'GET')
+          expect(spans[1].meta).to.have.property('http.status_code', '500')
+          expect(spans[1].meta).to.have.property('error', 'true')
+          }).then(done)
+          .catch(done)
+
+        this.server.listen('localhost', port, () => {
+        axios
+          .get(`http://localhost:${port}/api/user`)
```

Comment:
Mind catching the error to avoid logging event in the test run?

Reply:
Not quite sure how to do that. Using a try-catch still ended up logging the event in the test run.

Reply:
Never mind. Fixed it. It no longer logs errors.

Reply:
Never mind, fixed it. It no longer logs the test error events.